src/NHapi.Base/Model/GenericPrimitive.cs

```diff
@@ -1,47 +1,45 @@
-using System;
-
 namespace NHapi.Base.Model
 {
-   /// <summary> An unspecified Primitive datatype that imposes no constraints on its string 
-   /// value.  This is used to store Varies data, when the data type is unknown.  It is also 
-   /// used to store unrecognized message constituents.  
-   /// </summary>
-   /// <author>  Bryan Tripp
-   /// </author>
-   public class GenericPrimitive : AbstractPrimitive, IPrimitive
-	{
-		/// <summary> Returns a String representation of the value of this field.</summary>
-		/// <summary> Sets the value of this field if the given value is legal in the context of the
-		/// implementing class.
-		/// </summary>
-		/// <throws>  DataTypeException if the given value is not valid in this context. </throws>
-		public override String Value
-		{
-			get { return value_Renamed; }
+    /// <summary> An unspecified Primitive datatype that imposes no constraints on its string
+    /// value.  This is used to store Varies data, when the data type is unknown.  It is also
+    /// used to store unrecognized message constituents.
+    /// </summary>
+    /// <author>  Bryan Tripp.
+    /// </author>
+    public class GenericPrimitive : AbstractPrimitive, IPrimitive
+    {
+        private string valueRenamed = null;
 
-			set { value_Renamed = value; }
-		}
+        /// <summary> Creates a new instance of GenericPrimitive. </summary>
+        public GenericPrimitive(IMessage message)
+            : base(message)
+        {
+        }
 
-		/// <summary>Returns the name of the type (used in XML encoding and profile checking)  </summary>
-		public override String TypeName
-		{
-			get { return "UNKNOWN"; }
-		}
+        /// <summary> Returns a String representation of the value of this field.</summary>
+        /// <summary> Sets the value of this field if the given value is legal in the context of the
+        /// implementing class.
+        /// </summary>
+        /// <throws>  DataTypeException if the given value is not valid in this context. </throws>
+        public override string Value

```

Comment:
this can be simplified I think to...
```csharp
public override string Value { get; set; }
```
not sure why the field is neaded.

Reply:
done